test/close_socket_by_other_party.js

```diff
@@ -9,6 +9,31 @@ var connect = helper.connect
 var subscribe = helper.subscribe
 var delay = helper.delay
 
+test('aedes is closed before client authenticate returns', function (t) {
+  t.plan(0)
+
+  var broker = aedes({
+    authenticate: async (client, username, password, done) => {
+      await delay(2000) // simulate network
+      done(null, true)
+    }
```

Comment:
Please do not use an async function. It's a bad pattern because there will be no one checking for an error message, resulting in an unhandledRejection. CHeck out https://github.com/mcollina/make-promises-safe for more details.

Reply:
Does it mean that we also should not use async at here https://github.com/moscajs/aedes/blob/master/test/connect.js#L333 ?

Reply:
exactly. None of the aedes-codebase if promise friendly. We should possibly add support for those in the future.

Reply:
sounds like we need to replace the delay with setTimeout in this stage.
Which ways do you prefer to make aedes-codebase be promis friendly?

Reply:
open a new issue to discuss!